src/main/c/ssl.c

```diff
@@ -1366,59 +1476,83 @@ TCN_IMPLEMENT_CALL(jlong, SSL, makeNetworkBIO)(TCN_STDARGS,
     BIO *internal_bio;
     BIO *network_bio;
 
-    UNREFERENCED(o);
-
     if (ssl_ == NULL) {
         tcn_ThrowException(e, "ssl is null");
-        goto fail;
+        return 0;
     }
 
     if (BIO_new_bio_pair(&internal_bio, 0, &network_bio, 0) != 1) {
         tcn_ThrowException(e, "BIO_new_bio_pair failed");
-        goto fail;
+        return 0;
     }
 
+    UNREFERENCED(o);
+
     SSL_set_bio(ssl_, internal_bio, internal_bio);
 
     return P2J(network_bio);
- fail:
-    return 0;
 }
 
 // Free a BIO * (typically, the network BIO)
 TCN_IMPLEMENT_CALL(void, SSL, freeBIO)(TCN_STDARGS,
                                        jlong bio /* BIO * */) {
-    BIO *bio_;
+    BIO *bio_ = J2P(bio, BIO *);
+
+    if (bio_ == NULL) {
+        tcn_ThrowException(e, "bio is null");
+        return;
+    }
+
     UNREFERENCED_STDARGS;
 
-    bio_ = J2P(bio, BIO *);
     BIO_free(bio_);
 }
 
 // Send CLOSE_NOTIFY to peer
 TCN_IMPLEMENT_CALL(jint /* status */, SSL, shutdownSSL)(TCN_STDARGS,
                                                         jlong ssl /* SSL * */) {
+    SSL *ssl_ = J2P(ssl, SSL *);
+
+    if (ssl_ == NULL) {
+        tcn_ThrowException(e, "ssl is null");
+        return 0;
+    }
+
     UNREFERENCED_STDARGS;
 
-    return SSL_shutdown(J2P(ssl, SSL *));
+    return SSL_shutdown(ssl_);
 }
 
 // Read which cipher was negotiated for the given SSL *.
 TCN_IMPLEMENT_CALL(jstring, SSL, getCipherForSSL)(TCN_STDARGS,
                                                   jlong ssl /* SSL * */)
 {
+    SSL *ssl_ = J2P(ssl, SSL *);
+
+    if (ssl_ == NULL) {
+        tcn_ThrowException(e, "ssl is null");
+        return AJP_TO_JSTRING("");
```

Comment:
@nmittler NULL is ok as we throw the exception anyway and it is just to return early.